kong/openssl.patch

```diff
@@ -0,0 +1,274 @@
+-        ":install-openssl",
+     ],
+ )
+ 
+diff --git a/build/luarocks/BUILD.luarocks.bazel b/build/luarocks/BUILD.luarocks.bazel
+index b23ad96..d9dda79 100644
+--- a/build/luarocks/BUILD.luarocks.bazel
++++ b/build/luarocks/BUILD.luarocks.bazel
+@@ -40,7 +40,6 @@ kong_template_genrule(
+     name = "luarocks_exec",
+     srcs = [
+         "@libexpat",
+-        "@openssl",
+     ] + select({
+         "@kong//:any-cross": ["@cross_deps_libyaml//:libyaml"],
+         "//conditions:default": [
+diff --git a/build/luarocks/templates/luarocks_exec.sh b/build/luarocks/templates/luarocks_exec.sh
+index 10515f8..f8cc445 100644
+--- a/build/luarocks/templates/luarocks_exec.sh
++++ b/build/luarocks/templates/luarocks_exec.sh
+@@ -3,7 +3,6 @@
+ # template variables starts
+ libexpat_path="{{@@libexpat//:libexpat}}"
+ libxml2_path="invalid"
+-openssl_path="{{@@openssl//:openssl}}"
+ luarocks_host_path="{{@@luarocks//:luarocks_host}}"
+ luajit_path="{{@@openresty//:luajit}}"
+ kongrocks_path="invalid"
+@@ -27,7 +26,6 @@ ROCKS_CONFIG=$(readlink -f "$ROCKS_DIR/../luarocks_config.lua")
+ 
+ EXPAT_DIR=$root_path/$libexpat_path
+ LIBXML2_DIR=$root_path/$libxml2_path
+-OPENSSL_DIR=$root_path/$openssl_path
+ 
+ # The Bazel rules doesn't export the `libexpat.so` file,
+ # it only exports something like `libexpat.so.1.6.0`,
+@@ -115,8 +113,6 @@ fi
+ # some distros has BINPRM_BUF_SIZE smaller than the shebang generated,
+ # which is usually more than 160 bytes
+ $host_luajit $root_path/$LUAROCKS_HOST/bin/luarocks \$private_rocks_args \$@ \\
+-    OPENSSL_DIR=$OPENSSL_DIR \\
+-    CRYPTO_DIR=$OPENSSL_DIR \\
+     EXPAT_DIR=$EXPAT_DIR \\
+     LIBXML2_DIR=$LIBXML2_DIR \\
+     YAML_DIR=$YAML_DIR
+diff --git a/build/openresty/BUILD.openresty.bazel b/build/openresty/BUILD.openresty.bazel
+index 81ad172..157f6b7 100644
+--- a/build/openresty/BUILD.openresty.bazel
++++ b/build/openresty/BUILD.openresty.bazel
+@@ -147,10 +147,8 @@ CONFIGURE_OPTIONS = [
+     "--without-http_rds_csv_module",
+     "--with-luajit=$$EXT_BUILD_DEPS/luajit",
+     "--with-cc-opt=\"-I$$EXT_BUILD_DEPS/pcre/include\"",
+-    "--with-cc-opt=\"-I$$EXT_BUILD_DEPS/openssl/include\"",
+     "--with-cc-opt=\"-I$$EXT_BUILD_DEPS/luajit/include\"",
+     "--with-ld-opt=\"-L$$EXT_BUILD_DEPS/pcre/lib\"",
+-    "--with-ld-opt=\"-L$$EXT_BUILD_DEPS/openssl/lib\"",
+     "--with-ld-opt=\"-L$$EXT_BUILD_DEPS/luajit/lib\"",
+     "--with-ld-opt=\"-L$$EXT_BUILD_DEPS/lib\"",
+     # Here let's try not having --disable-new-dtags; --disable-new-dtags creates rpath instead of runpath
+@@ -326,7 +324,6 @@ configure_make(
+     visibility = ["//visibility:public"],
+     deps = [
+         "@openresty//:luajit",
+-        "@openssl",
+         "@pcre",
+     ] + select({
+         "@kong//:any-cross": [
+diff --git a/build/openresty/openssl/BUILD.bazel b/build/openresty/openssl/BUILD.bazel
+deleted file mode 100644
+index 5970c67..0000000
+--- a/build/openresty/openssl/BUILD.bazel
```

Comment:
Exclude this portion of the patch and do this in the package pipeline: `rm -r build/openresty/openssl`